src/bitcoind/interface.rs

```diff
@@ -599,46 +569,113 @@ impl BitcoinD {
+                } else {
+                    None
+                }
+            })
+            .collect())
+    }
+
+    pub fn list_deposits_since_block(
+        &self,
+        tip: &BlockchainTip,
+    ) -> Result<Vec<ListSinceBlockTransaction>, BitcoindError> {
+        self.list_since_block(tip, Some(DEPOSIT_UTXOS_LABEL))
+    }
+
+    pub fn list_unvaults_since_block(
+        &self,
+        tip: &BlockchainTip,
+    ) -> Result<Vec<ListSinceBlockTransaction>, BitcoindError> {
+        self.list_since_block(tip, Some(UNVAULT_UTXOS_LABEL))
+    }
+
     /// Repeatedly called by our main loop to stay in sync with bitcoind.
     /// We take the currently known deposit utxos, and return the new, confirmed and spent ones.
     pub fn sync_deposits(
         &self,
         deposits_utxos: &HashMap<OutPoint, UtxoInfo>,
+        db_tip: &BlockchainTip,
         min_conf: u32,
     ) -> Result<DepositsState, BitcoindError> {
-        let (mut new_utxos, mut confirmed_utxos) = (HashMap::new(), HashMap::new());
-        // All seen utxos, if an utxo remains unseen by listunspent then it's spent.
-        let mut spent_utxos = deposits_utxos.clone();
-        let utxos = self.list_unspent_deposits(Some(MIN_DEPOSIT_VALUE / COIN_VALUE))?;
-
-        for unspent in utxos {
-            // Not obvious at first sight:
-            //  - spent_utxos == existing_utxos before the loop
-            //  - listunspent won't send duplicated entries
-            //  - remove() will return None if it was not present in the map
-            // Therefore if there is an utxo at this outpoint, it's an already known deposit
-            if let Some(map_utxo) = spent_utxos.remove(&unspent.outpoint) {
-                // It may be known but still unconfirmed, though.
-                if !map_utxo.is_confirmed && unspent.confirmations >= min_conf as u64 {
-                    confirmed_utxos.insert(unspent.outpoint, map_utxo);
+        let (mut new_unconf, mut new_conf, mut new_spent) =
+            (HashMap::new(), HashMap::new(), HashMap::new());
+
+        // First, we check the existing deposits to see whether the unconfirmed got confirmed or if
+        // any was spent.
+        // FIXME: batch those calls to gettxout
+        for (outpoint, info) in deposits_utxos {
+            let confirmations = self.get_unspent_outpoint_confirmations(&outpoint)?;
+            if let Some(confirmations) = confirmations {
+                if !info.is_confirmed && confirmations >= min_conf as i32 {
+                    new_conf.insert(
+                        *outpoint,
+                        UtxoInfo {
+                            txo: info.txo.clone(),
+                            is_confirmed: true,
+                        },
+                    );
+                }
+            } else {
+                // We've seen it unspent and now it's not here anymore: it's spent.
+                new_spent.insert(*outpoint, info.clone());
+            }
```

Comment:
Note so i don't forget: this actually introduces the possibility for a vault to go from `unconfirmed` to `unvaulting` immediately.

I'm not sure how we should approach to patch this yet.